### PR TITLE
[1.20.x] Pull `EntityEvent$Size` changes from MCF/1.20.x

### PR DESCRIPTION
**Pull request by Shadows-of-Fire**
This PR is a subset of #14 which pulls in all commits relating to `EntityEvent$Size`, and https://github.com/MinecraftForge/MinecraftForge/pull/9428 which depends on the commit history of those changes.

### CODE REVIEW

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 1]
```diff
@@ -772,20 +769,6 @@ public static void onCommandRegister(CommandDispatcher<CommandSourceStack> dispa
...
-        MinecraftForge.EVENT_BUS.post(evt);
-        return evt;
-    }
-
-    public static net.minecraftforge.event.entity.EntityEvent.Size getEntitySizeForge(Entity entity, Pose pose, EntityDimensions oldSize, EntityDimensions newSize, float newEyeHeight)
-    {
-        EntityEvent.Size evt = new EntityEvent.Size(entity, pose, oldSize, newSize, entity.getEyeHeight(), newEyeHeight);
-        MinecraftForge.EVENT_BUS.post(evt);
-        return evt;
-    }
```
Comment (pupnewfster) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java:787]: These two methods should be reinstated and deprecated for removal as removing them is a breaking change (and also definitely [affects](https://gist.github.com/Pewmaymen/2d9ae4bacb9a128aae8f914f8008b010) mods)

**Review by pupnewfster — CHANGES_REQUESTED** [review]

**Comment by mcg-anon** [comment]
> [CLA assistant check](https://cla-assistant.io/neoforged/NeoForge?pullRequest=22) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/neoforged/NeoForge?pullRequest=22) before we can accept your contribution.**0** out of **4** committers have signed the CLA.x Meldexunx LexManosx ChampionAsh5357x JonathingYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/neoforged/NeoForge?pullRequest=22) it.

I don't think that LexManos is going to sign your CLA, just saying.

**Comment by TheCurle** [comment]
This looks like it could solve one of the issues I was having with gravity wells in one of my mods.
However, those commits will cause a problem with our CLA. We should try to reimplement them in a new (better!) way. And also fix those merge conflicts.

I'll leave this open until I (or someone else) gets around to that.

**Comment by pupnewfster** [comment]
I believe the entire event will likely become redundant in 1.20.5, so may not be worth the effort.

**Comment by TheCurle** [comment]
Do note that this is (was?) a 1.20.1 PR. It should be retargeted as such, if 1.20.5 does fix these issues.

**Comment by pupnewfster** [comment]
No I mean mojang is adding a scale attribute to 1.20.5 so when that releases I doubt we will have a use case for this event.

**Comment by Shadows-of-Fire** [comment]
Yeah I think we can just toss this one in favor of the scale attribute.  Its all the size event was for anyway.

**Comment by TheCurle** [comment]
We will; in 1.20.1, where this PR lives. It targets the 1.20.x branch because there was no 1.20.1 branch at its time of creation.

**Comment by TheCurle** [comment]
That works too. :P

**Pull request closed without merging**